keepalived/check/check_misc.c

```diff
@@ -71,6 +71,26 @@ dump_misc_check(void *data)
 	log_message(LOG_INFO, "   insecure = %s", misck_checker->insecure ? "Yes" : "No");
 }
 
+static int
+compare_misc_check(void *a, void *b)
+{
+	misc_checker_t *old = CHECKER_DATA(a);
+	misc_checker_t *new = CHECKER_DATA(b);
+
+	if (strcmp(old->path, new->path) != 0)
+		goto err;
+	if (old->timeout != new->timeout)
+		goto err;
+	if (old->dynamic != new->dynamic)
+		goto err;
+	if (old->uid != new->uid || new->gid != new->gid)
+		goto err;
```

Comment:
I'm not sure that different values of uid or gid make this a different checker; however it is possible that the script could take different actions depending on its uid. My inclination is that different uid/gid shouldn't make this a different checker.